tests/functional/parser/test_sqs_s3_event_notification.py

```diff
@@ -0,0 +1,43 @@
+import pytest
+
+from aws_lambda_powertools.utilities.parser import ValidationError, event_parser
+from aws_lambda_powertools.utilities.parser.models import SqsS3EventNotificationModel
+from aws_lambda_powertools.utilities.typing import LambdaContext
+from tests.functional.parser.test_s3 import assert_s3
+from tests.functional.utils import json_serialize, load_event
+
+
+@event_parser(model=SqsS3EventNotificationModel)
+def handle_sqs_json_body_containing_s3_notifications(event: SqsS3EventNotificationModel, _: LambdaContext):
+    return event
+
+
+def test_handle_sqs_json_body_containing_s3_notifications():
+    sqs_event_dict = load_event("sqsEvent.json")
+    s3_event_notification_dict = load_event("s3Event.json")
+    for record in sqs_event_dict["Records"]:
+        record["body"] = json_serialize(s3_event_notification_dict)
+
+    parsed_event: SqsS3EventNotificationModel = handle_sqs_json_body_containing_s3_notifications(
+        sqs_event_dict, LambdaContext()
+    )
+
+    assert len(parsed_event.Records) == 2
+    for parsed_sqs_record in parsed_event.Records:
+        assert_s3(parsed_sqs_record.body)
```

Comment:
I would like an opinion here @rubenfonseca @heitorlessa. I think this test is ok, do you see another way to optimize this?

Reply:
Looks ok to me

Reply:
This is a good opportunity to move to the new unit tests since all we should care here is whether the new Model can be instantiated correctly (along with its ramifications, JSON deserialization etc.).

Following this, all you'd need to do here is:

* Create a new S3->SQS JSON event file under `tests/events` 
* Create a new test at `tests/unit/parser/test_s3.py`
* Test whether the model can be initialized without any problem (we already test both SQS and S3 Models)
* Test whether an invalid JSON in the S3 data will lead to a model validation failure

Previously, we were shoving everything under `tests/functional` where we have a lot of boilerplate like `@event_parser` and these functions, when in fact we're merely testing the Model since we already have separate tests for the `@event_parser`... which in this case is a proper `functional` test.

Hope that makes sense ;)

Reply:
It makes perfect sense @heitorlessa! I changed the files as per this feedback